modules/openideal_idea/src/Plugin/Block/OpenidealIdeaUpdateInfo.php

```diff
@@ -110,10 +121,15 @@ public function build() {
         'access' => $this->configuration['use_updated'],
       ];
 
-      $member = $this->helper->getGroupMember($this->currentUser, $node);
-      if ($member && $member->hasPermission('update any group_node:idea entity')) {
-        $link = Link::createFromRoute($this->t('Edit'), 'entity.node.edit_form', ['node' => $node->id()])->toString();
-        $build['#content']['edit'] = $link;
+      $group = $this->helper->getGroupFromNode($node);
+      if (($group && $this->groupPermissionChecker->hasPermissionInGroup('update any group_node:idea content', $this->currentUser, $group)) ||
+        (!$group && $node->access('update', $this->currentUser))) {
+        $link = Link::createFromRoute($this->t('Edit'), 'entity.node.edit_form', ['node' => $node->id()])->toString()->getGeneratedLink();
+        $build['#content']['edit'] = [
```

Comment:
What is the cache strategy here? Do we have any cache context?

Reply:
This is our default cache strategy, if no node was provided then we can't add any cache tags/context, otherwise, we add cache tags from the node, and cache contexts route.